### PR TITLE
Feature/external login providers

### PR DESCRIPTION
**Pull request by patrikulus**
Added possibility to log in by OAuth providers

## Description
We have google, facebook, twitter and github providers.

## Where should the reviewer start?
Check if the buttons are redirecting to external providers. The provider's sites will show error but that's ok because there are no secrets in appsettings.json file.

## Motivation and context
#60 

## How has this been tested?
All providers were checked by doing these three steps:
- changed redirect-url to localhost in provider's application configuration
- replaced secret placeholders to real values
- performed register by given provider

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by WTobor** [comment]
Buttons are not looking good on small screen; I suggest one button under another with class col-xs-4
image

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Services/AccountService.cs [hunk 1]
```diff
@@ -35,7 +35,10 @@ public string GetUserId(ClaimsPrincipal principal)
 
         public async Task<IdentityResult> CreateAsync(ApplicationUser user, string password)
         {
-            return await _userManager.CreateAsync(user, password);
+            if (string.IsNullOrEmpty(password) || password == "dupa")
```
[the remark this answers is not in the document]
Comment (patrikulus) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Services/AccountService.cs]: shame on me :(

[the remark this answers is not in the document]
Comment (patrikulus) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Services/AccountService.cs]: Fixed

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Account/Register.cshtml [hunk 2]
```diff
@@ -30,6 +34,39 @@
     </div>
 </div>
 
+<br/>
+
+@{
+    var externalProviders = await IdentityService.GetExternalAuthenticationSchemes();
+
+    if (externalProviders != null && externalProviders.Count == 0)
```
[the remark this answers is not in the document]
Comment (patrikulus) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Account/Register.cshtml]: I see this is fixed already ;)

**Pull request merged**